src/SFA.Tl.Matching.Automation.Tests/Project/Framework/Helpers/PageInteractionHelper.cs

```diff
@@ -188,5 +235,133 @@ public static String GetText(By locator)
+                        
+            var queryResults = SqlDatabaseConncetionHelper.ReadDataFromDataBase(query, Configurator.GetConfiguratorInstance().GetMathcingServiceConnectionString());
+
+            String Name;
+            String Postcode;
+            double Latitude;
+            double Longitude;
+            int providerCount = 0;
+
+            foreach (object[] fieldNo in queryResults)
+            {
+                //Assign values to variables from the SQL query run
+                Name = fieldNo[0].ToString();
+                Postcode = (fieldNo[1].ToString());
+                Latitude = Convert.ToDouble(fieldNo[2]);
+                Longitude = Convert.ToDouble(fieldNo[3]);
+
+                //longitude and latitude values for postcode B43 6JN
+                double postcodeLong = -1.93253;
+                double postcodelat = 52.54834;
+                //Calculate the distance of each postcode from postcode B43 6JN
+                double distanceFromPostcode = PageInteractionHelper.DistanceBetweenPlaces(Longitude, Latitude, postcodeLong, postcodelat);
+                
+                //if the postcode is within 25 miles of B43 6JN, verify the provider name and postcode is displayed on screen               
+                if (distanceFromPostcode <= _postcodeRadius)
+                {
+                    Console.WriteLine("Distance is less than " + _postcodeRadius + " miles. Provider name is " + Name + " " + Postcode + " distance is  " + distanceFromPostcode);
+                    providerCount = providerCount +1;
+
+                    VerifyProviderDisplayed(Name);
+                    VerifyProviderPostcodeDisplayed(Postcode);
+                }            
+            }
+            ScenarioContext.Current["SearchResultsCount"] = providerCount;
+        }
+
+        public static void SetProviderNames(By locator1, By locator2)
+        {
+            IWebElement webElement1 = webDriver.FindElement(locator1);
+            String text1 = webElement1.Text.Split('\r')[0];
+            text1 = text1.TrimEnd();
+            Console.WriteLine("Provider 1" + text1);
+            ScenarioContext.Current["_Provider1"] = text1;
+            
+            IWebElement webElement2 = webDriver.FindElement(locator2);
+            String text2 = webElement2.Text.Split('\r')[0];
+            text2 = text2.TrimEnd();
+            Console.WriteLine("Provider 2" + text2);
+            ScenarioContext.Current["_Provider2"] = text2;
+        }
```

Comment:
All the code from Line-251 does not belong in this class. This is only a generic helper class at framework level, anything specific to the application should be defined some where else. Please review